bson/src/main/org/bson/BsonBinaryWriter.java

```diff
@@ -37,8 +37,16 @@ public class BsonBinaryWriter extends AbstractBsonWriter {
 
     private final BsonOutput bsonOutput;
     private final Stack<Integer> maxDocumentSizeStack = new Stack<>();
+    private static final int ARRAY_INDEXES_CACHE_SIZE = 1 << 8;
```

Comment:
Setting it to 256 explicitly reduces the cognitive load.

```suggestion
    private static final int ARRAY_INDEXES_CACHE_SIZE = 256;
```